programmers/이상한-문자-만들기.js

```diff
@@ -0,0 +1,19 @@
+function solution(s) {
+    let splitS = s.split(' ');
+    let splitWord;
+    const answer = [];
+    
+    for (let el of splitS) { 
+        splitWord = el.split('')
+        
+        for (let i = 0; i < splitWord.length; i++) {
+            if (i % 2 === 0) splitWord[i] = splitWord[i].toUpperCase();
+            else splitWord[i] =  splitWord[i].toLowerCase();  
+        }
+        
+        answer.push(splitWord.join(''));
+    }
+    
+    const result = answer.join(' ');
+    return result;
```

Comment:
천천히 짚어보자면
1. const가 let보다 상단에 위치해야합니다.
2.  파라미터 넘어오는 이름이 s여서 변수명이 좋기 힘듭니다 그럴땐 for문에 바로 넣어도 괜찮습니다.
3. el 이라고 요소 이름을 짓는것은 지양합니다. 
4. 길이를 재기 위해  splitWord 를 선언한 것 같은데 선언하지 않고 바로 넣어도 됩니다
5. if 문 대신 삼항을 사용하였는데 if문을 더 선호하는 회사도 있다고 합니다. 그런데 프론트 개발을 하다보면 어쩔 수 없이 삼항만을 사용해야하는 상황이 옵니다 그럴때를 대비해서 삼항도 조금은 익혀두면 좋을거 같아요
6. push를 사용하지 않고 공백을 넣어줍니다.
공백을 넣어주면 맨마지막에도 공백이 붙기에 slice를 사용해서 공백을 잘라주면됩니다.
```suggestion
    let convertedWord  = "";
    
    for (let word of s.split(' ')) { 
        for (let i = 0; i < word.length; i++) {
            convertedWord += i % 2 ? word[i].toLowerCase() : word[i].toUpperCase(); 
        }
        convertedWord += " ";
    }
    return convertedWord.slice(0, convertedWord.length - 1);
```